impl/src/planning.rs

```diff
@@ -568,26 +568,26 @@ impl<'planner> CrateSubplanner<'planner> {
     for dep in &package.dependencies {
       if dep.target.is_some() {
         // UNWRAP: Safe from above check
-        let target_str = dep.target.as_ref().unwrap();
-        let platform = Platform::from_str(target_str)?;
+        let target_str = format!("{}", dep.target.as_ref().unwrap());
+        let platform = Platform::from_str(&target_str)?;
 
         // Skip this dep if it doesn't match our platform attributes
         if !platform.matches(&self.settings.target, platform_attrs.as_ref()) {
           continue;
         }
       }
 
-      match dep.kind.as_deref() {
-        None | Some("normal") => normal_dep_names.push(dep.name.clone()),
-        Some("dev") => dev_dep_names.push(dep.name.clone()),
-        Some("build") => build_dep_names.push(dep.name.clone()),
-        something_else => {
+      match dep.kind {
+        DependencyKind::Normal => normal_dep_names.push(dep.name.clone()),
```

Comment:
Just checking, is this equivalent to the old code? Where "no" dependency kind results in inferring normal?

I think this is really a question of what cargo_metadata returns in that case.

(again, I think the tests are not covering this case....)